ci: add mergify rule for release 3.6 backport

Madhu-1 opened the pull request:
added mergify rules to create the backport PR for release-v3.6 branch.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 committed:
ci: add mergify rule for 3.6 backport

added mergify rules to create the backport
PR for release-3.6 branch.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

nixpanic reviewed (changes requested):
With 3.6 released, there is no need to keep the 3.4 rules. Please remove them as well.

Madhu-1 committed:
ci: remove release-v3.4 mergify rules

as we have deprecated the 3.4 release
removing the mergify rules for the same.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 committed:
ci: remove duplicate release-3.5 merge rules

we already have generic rules to merge the PR's
in devel and release branches with `automatic merge`
rules. Removing the duplicate release-3.5 rule.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 commented:
> With 3.6 released, there is no need to keep the 3.4 rules. Please remove them as well.

Done removed duplicate 3.5 and 3.4 rules

Madhu-1 commented:
Need manual merge. Merging it.
